modules/cloudsql-instance/main.tf

```diff
@@ -289,8 +289,22 @@ resource "random_password" "passwords" {
     k
     if v.password == null
   ])
-  length  = 16
-  special = true
+  length      = try(var.password_validation_policy.min_length, 16)
+  special     = true
+  min_lower   = 1
+  min_numeric = 1
+  min_special = 1
+  min_upper   = 1
+}
+
+resource "random_password" "root_password" {
+  count       = var.root_password == null ? 1 : 0
```

Comment:
Should we bother using count to prevent this resource being created (and not used) in some cases?  It's only a local resource, and means that we need to use `[0]` when accessing it.

Reply:
No issues using count if you don't need the password. It's actually a pretty common pattern.